service/src/main/java/org/cbioportal/genome_nexus/service/internal/VerifiedGenomicLocationAnnotationServiceImpl.java

```diff
@@ -161,14 +161,31 @@ private VariantAnnotation verifyOrFailAnnotation(VariantAnnotation annotation)
         return createFailedAnnotation(originalVariantQuery, originalVariant);
     }
 
+    private Boolean needFollowUpQuery(String responseReferenceAllele, String providedReferenceAllele)
+    {
+        // for altered length responses, we need to recover full reference allele from follow up query
+        // follow up query gives correct reference allele for referenced genome positions
+        if (responseReferenceAllele.length() != providedReferenceAllele.length()) {
+            // for altered length deletions, e.g. '1,123,124,TC,T': responseReferenceAllele='C', providedReferenceAllele='TC', length not equal
+            // for altered length del-ins, e.g. '1,123,124,TC,TAA': responseReferenceAllele='C', providedReferenceAllele='TC', length not equal
+            // for altered length insertion but > 1 matching alleles, e.g. '1,123,123,TC,TCA': responseReferenceAllele='-', providedReferenceAllele='TC', length not equal
+            return true;
+        }
+        else if(responseReferenceAllele.length() == providedReferenceAllele.length() && responseReferenceAllele.equals("-")) {
```

Comment:
The conditional checks here seem redundant. I think the code is equivalent to:
```
if (not X) {
    ...;
} else if (X) {
    if (Y) {
        ...;
    }
}
```

Also, the function exits at the first return statement, so we can simplify the logic I think:
```
if (lengths_not_equal) {
    return true;
}
if (responseReferenceAllele.equals("-") {
    return true;
}
return false;
```
Similar to not having an else for the final "return false", we don't need an else for the second test and we know that the lengths are equal (or we would have exited during the first test).

Also please add a space after the keyword "if".